### PR TITLE
Added support to use tolerations for head and worker nodes

### PR DESCRIPTION
**Pull request by navneet066**
## Why are these changes needed?

As Ray deals in GPU nodes also, there will be mostly Taint present in GPU nodes. So we need to have a toleration section to sepcify in the `values.yaml` file.

I have made a couple of more minor changes regarding the `imagePullSecrets` as well

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by edoakes** [comment]
@DmitriGekhtman could you please review this?

File: deploy/charts/ray/values.yaml [hunk 1]
```diff
@@ -83,3 +108,10 @@ operatorImage: rayproject/ray:latest
 # For a particular official release version of Ray, use `rayproject/ray:1.x.y`.
 # For a specific master commit, use the first 6 characters of the commit SHA, e.g. `rayproject/ray:050a07`.
 # The operator and Ray cluster can use different Ray versions, provided both versions are >= 1.2.0
+
+## imagePullSecret for the rey operator
+## Reference to one or more secrets to be used when pulling images
+## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+imagePullSecrets: []
```
Comment (DmitriGekhtman) [inline deploy/charts/ray/values.yaml:115]: I think we should move operator settings into an operator subfield, while keeping compatibility with the current chart -- I can help out with this.  

I like the change of having the pull secrets be a per-pod setting.

**Comment by DmitriGekhtman** [comment]
Thanks! These are valuable contributions. I think it would be good if we could collaborate on this PR -- @navneet066 would you mind if I push commits to your PR branch?

**Comment by navneet066** [comment]
Yeah Sure. You can do that. Main Idea is to make is more robust.

Thanks

**Comment by navneet066** [comment]
> Thanks! These are valuable contributions. I think it would be good if we could collaborate on this PR -- @navneet066 would you mind if I push commits to your PR branch?

I have invited you as a collaborator for the repo. You can make the changes :)

**Comment by navneet066** [comment]
Hi @DmitriGekhtman Just wanted to check if you had a chance to look into this PR?

**Comment by DmitriGekhtman** [comment]
> Hi @DmitriGekhtman Just wanted to check if you had a chance to look into this PR?

Yep -- I've started working on it, but unfortunately got tied up with other things. I'll aim to push changes by the end of the week. 
In the meantime, feel free of course to work with the version of the chart in the PR branch.

**Comment by navneet066** [comment]
@DmitriGekhtman Just to let you know, I have removed the imagePullSecret part from this PR, As you mentioned here https://github.com/ray-project/ray/pull/17520#issuecomment-895346882

**Comment by DmitriGekhtman** [comment]
Got it, thanks.

**Comment by DmitriGekhtman** [comment]
This PR looks good -- I just pushed some updates to the documenting comments in `values.yaml`.

**Review by DmitriGekhtman — APPROVED** [review]

**Comment by DmitriGekhtman** [comment]
@edoakes 
Since I added some stuff to this PR -- for sake of good process, could you take a quick look and approve if it looks good?

**Review by edoakes — APPROVED** [review]

**Comment by DmitriGekhtman** [comment]
mac failures definitely unrelated, mergin

**Pull request merged**